channels/serializers.py

```diff
@@ -126,6 +126,10 @@ def validate_banner(self, value):
             raise ValidationError("Expected banner to be a file")
         return {"banner": value}
 
+    def validate_membership_is_managed(self, value):
```

Comment:
Is this needed? I see other `WriteableSerializerMethodField` fields that don't have a corresponding `validate_` method. Not seeing why this is necessary here

Reply:
Yes, the channel creation request will fail without it if `membership_is_managed` is sent as part of the request body:

```
web_1      | [2018-10-31 21:56:41] ERROR 44 [django.request] log.py:228 - [a7c9a84348b0] - Internal Server Error: /api/v0/channels/
web_1      | Traceback (most recent call last):
web_1      |   File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
web_1      |     response = get_response(request)
web_1      |   File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 126, in _get_response
web_1      |     response = self.process_exception_by_middleware(e, request)
web_1      |   File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 124, in _get_response
web_1      |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
web_1      |   File "/usr/local/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
web_1      |     return view_func(*args, **kwargs)
web_1      |   File "/usr/local/lib/python3.6/site-packages/django/views/generic/base.py", line 68, in view
web_1      |     return self.dispatch(request, *args, **kwargs)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 494, in dispatch
web_1      |     response = self.handle_exception(exc)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 454, in handle_exception
web_1      |     self.raise_uncaught_exception(exc)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 491, in dispatch
web_1      |     response = handler(request, *args, **kwargs)
web_1      |   File "./channels/views/channels.py", line 54, in post
web_1      |     return super().post(request, *args, **kwargs)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/generics.py", line 244, in post
web_1      |     return self.create(request, *args, **kwargs)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/mixins.py", line 20, in create
web_1      |     serializer.is_valid(raise_exception=True)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/serializers.py", line 236, in is_valid
web_1      |     self._validated_data = self.run_validation(self.initial_data)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/serializers.py", line 435, in run_validation
web_1      |     value = self.to_internal_value(data)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/serializers.py", line 475, in to_internal_value
web_1      |     set_value(ret, field.source_attrs, validated_value)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/fields.py", line 125, in set_value
web_1      |     dictionary.update(value)
web_1      | TypeError: 'bool' object is not iterable
web_1      | [2018-10-31 21:56:41] ERROR 44 [django.request] log.py:228 - [a7c9a84348b0] - Internal Server Error: /api/v0/channels/
web_1      | Traceback (most recent call last):
web_1      |   File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
web_1      |     response = get_response(request)
web_1      |   File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 126, in _get_response
web_1      |     response = self.process_exception_by_middleware(e, request)
web_1      |   File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 124, in _get_response
web_1      |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
web_1      |   File "/usr/local/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
web_1      |     return view_func(*args, **kwargs)
web_1      |   File "/usr/local/lib/python3.6/site-packages/django/views/generic/base.py", line 68, in view
web_1      |     return self.dispatch(request, *args, **kwargs)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 494, in dispatch
web_1      |     response = self.handle_exception(exc)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 454, in handle_exception
web_1      |     self.raise_uncaught_exception(exc)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/views.py", line 491, in dispatch
web_1      |     response = handler(request, *args, **kwargs)
web_1      |   File "./channels/views/channels.py", line 54, in post
web_1      |     return super().post(request, *args, **kwargs)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/generics.py", line 244, in post
web_1      |     return self.create(request, *args, **kwargs)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/mixins.py", line 20, in create
web_1      |     serializer.is_valid(raise_exception=True)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/serializers.py", line 236, in is_valid
web_1      |     self._validated_data = self.run_validation(self.initial_data)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/serializers.py", line 435, in run_validation
web_1      |     value = self.to_internal_value(data)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/serializers.py", line 475, in to_internal_value
web_1      |     set_value(ret, field.source_attrs, validated_value)
web_1      |   File "/usr/local/lib/python3.6/site-packages/rest_framework/fields.py", line 125, in set_value
web_1      |     dictionary.update(value)
web_1      | TypeError: 'bool' object is not iterable
```

Reply:
OK. The pattern in general seems to be kind of a code smell, but that's not an issue for this PR.